scripts/pre_commit_hook.py

```diff
@@ -139,13 +110,17 @@ def main():
         sys.exit(0)
 
     python_utils.PRINT('Running pre-commit check for package-lock.json ...')
-    if _does_diff_include_package_lock_file_and_no_package_file():
+    if _does_diff_include_package_lock_file():
+        # The following message is necessary since there git commit aborts
+        # quietly when the status is non-zero.
+        python_utils.PRINT('-----------COMMIT ABORTED-----------')
         python_utils.PRINT(
-            'This commit only changes package-lock.json without '
-            'any change in package.json. Therefore changes in '
-            'package-lock.json will be automatically reverted.')
-        python_utils.PRINT('Reverting changes in package-lock.json ...')
-        _revert_changes_in_package_lock_file()
+            'Oppia utilize Yarn to manage node packages. Please delete '
+            'package-lock.json, revert the changes in package.json, and use '
```

Comment:
If `package-lock.json` is not going to be a part of the repo then should we ignore that file? (I'm not sure how the yarn works.)

Reply:
nope, because we still want the pre commit hook to pick it up (if we ignore it, the pre commit hook won’t pick it up). good point though.

Reply:
> we still want the pre commit hook to pick it up

@kevinlee12 Why we still want pre-commit hook to pick up if that's not going to be a part of the repo?

Reply:
We want to pick it up so that we can alert the developer that they are using npm instead of yarn. We could technically use other ways to figure out whether they are using npm (like checking for the existence of package-lock.json), but it might have consequences down the line since I think yarn and npm treat package.json slightly differently in terms of dependency treatment.

Reply:
> We want to pick it up so that we can alert the developer that they are using npm

Pre-push hook will pick it up if a user will move `package-lock.json` file to tracked files, right? If yes, then we aren't doing what exacting we are expecting this method to do.

Also, if the motto is just to alert the contributors to use yarn, then it should be a part of start.sh or install related scripts, that way we can indicate them what they should use? I think pre-push hook should be for any code related changes (mostly for tracked files) not for adding alert related to what package to use.

Reply:
I can go ahead and add another check for the existence of `package-lock.json` so that we can be extra sure that they aren't using npm. 

As for letting users to know to use Yarn, I don't think we should put in start.sh. The output is very crowded there and it's easy to not see it. Putting a check in the pre commit check is there as another layer of defence. Once this is merged, I'll shoot out an email letting contributors know that we're switching to yarn for managing npm packages.

Reply:
^ I've added that check.

Reply:
`_does_diff_include_package_lock_file` the name is incorrect now.

Also, I'm still not sure why we are adding an untracked file check to pre_commit hook? pre_commit hook should be for testing the changes we are going to commit. I think we should move this check to the `install_third_party.sh` or `setup.sh` or any such file. This will also make sure that contributors/CI are going through this check before running test, starting server, etc.

> The output is very crowded there and it's easy to not see it

We can put it in the script where we use yarn to download modules and stop the script if they aren't using yarn. (Message should be at the end, we can use color if required.)

Reply:
> _does_diff_include_package_lock_file the name is incorrect now.

I found you have added a new method `_does_current_folder_contain_have_package_lock_file`, which does this new check! Ignore the first sentence of my last comment, but the `_does_diff_include_package_lock_file` method isn't doing anything related to the commit, right?

Reply:
Nope, it just prevents the commit from being done (via doing a `sys.exit(1)`). It'll be up to the contributor to take the necessary steps to undo the npm operation and do it again in yarn.

Reply:
@kevinlee12, I'm fine with the check in pre_commit hook, but I think it should be in the script file (Setup script) first and we can have this check inside pre_commit hook as well.

Reply:
I'll go ahead and add this message to the setup script.